### PR TITLE
Global ratelimiter: client-side Collection skeleton

### PR DESCRIPTION
**Pull request by Groxx**
A half-fleshed-out skeleton of the quotas.Collection that the frontend will use for these limits,
and a chunk of pseudo-code to show how it'll be used when the RPC PR is ready (coming soon).

This is not completely usable right now and will likely need some small changes, but it has the core bits:
- real limiters / fallbacks / tracking
- dynamically-tweakable background update loop
- logs, metrics, error handling (where reasonable)
- some very simple tests

It will eventually need:
- sharding the limiters by key for all the (relevant) aggregator hosts
- dynamic config to shadow / disable global limiters and control sending update requests.
- swapping the frontend handler's ratelimits to use this instead

### CODE REVIEW

File: common/quotas/global/collection/collection.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+) *Collection {
+	contents := internal.NewAtomicMap(func(key string) *internal.FallbackLimiter {
+		return internal.NewFallbackLimiter(fallback.GetLimiter(key))
+	})
+	ctx, cancel := context.WithCancel(context.Background())
+	c := &Collection{
+		limits:     contents,
+		updateRate: updateRate,
+
+		logger: logger.WithTags(tag.ComponentArchiver),
```
Comment (Shaddoll) [inline common/quotas/global/collection/collection.go:96]: nit: wrong tag

File: common/quotas/global/collection/collection.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+	// calls is a small temporary pair[T] container
+	calls struct {
+		allowed, rejected int
+	}
+)
+
+func New(
+	fallback quotas.LimiterFactory,
+	updateRate dynamicconfig.DurationPropertyFn,
```
Comment (Shaddoll) [inline common/quotas/global/collection/collection.go:84]: given the type, I prefer rename it to updateInterval

**Review by Shaddoll — APPROVED** [review]

File: common/quotas/global/collection/collection.go (hunk 1 above)
Comment (Groxx) [inline common/quotas/global/collection/collection.go:96]:
```suggestion
		logger: logger.WithTags(tag.ComponentGlobalRatelimiter),
```

Comment (Groxx) [reply common/quotas/global/collection/collection.go:96]: oops - thanks, will fix :)

File: common/quotas/global/collection/collection.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+func (c *Collection) OnStop(ctx context.Context) error {
+	c.ctxCancel()
+	// context timeout is for everything to shut down, but this one should be fast.
+	// don't wait very long.
+	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+	defer cancel()
+
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("*Collection failed to stop, context canceled: %w", ctx.Err())
```
Comment (dkrotx) [inline common/quotas/global/collection/collection.go:140]: nit: I think it is very unexpected to see pointer in error-log.

Comment (Groxx) [reply common/quotas/global/collection/collection.go:140]: huh.  yeah, I think that got caught in a refactor 🤔 
changing 👍

**Pull request merged**